main.py

```diff
@@ -0,0 +1,102 @@
+    parser.add_argument("output_folder", help="Path to the output folder")
+    parser.add_argument(
+        "--tts",
+        choices=get_supported_tts_providers(),
+        default=get_supported_tts_providers()[0],
+        help="Choose TTS provider (default: azure). azure: Azure Cognitive Services, openai: OpenAI TTS API. When using azure, environment variables MS_TTS_KEY and MS_TTS_REGION must be set. When using openai, environment variable OPENAI_API_KEY must be set.",
+    )
+    parser.add_argument(
+        "--log",
+        choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+        default="INFO",
+        help="Log level (default: INFO), can be DEBUG, INFO, WARNING, ERROR, CRITICAL",
+    )
+    parser.add_argument(
+        "--preview",
+        action="store_true",
+        help="Enable preview mode. In preview mode, the script will not convert the text to speech. Instead, it will print the chapter index, titles, and character counts.",
+    )
+    parser.add_argument(
+        "--language",
+        default="en-US",
+        help="Language for the text-to-speech service (default: en-US). For Azure TTS (--tts=azure), check https://learn.microsoft.com/en-us/azure/ai-services/speech-service/language-support?tabs=tts#text-to-speech for supported languages. For OpenAI TTS (--tts=openai), their API detects the language automatically. But setting this will also help on splitting the text into chunks with different strategies in this tool, especially for Chinese characters. For Chinese books, use zh-CN, zh-TW, or zh-HK.",
+    )
+    parser.add_argument(
+        "--newline_mode",
+        choices=["single", "double"],
+        default="double",
+        help="Choose the mode of detecting new paragraphs: 'single' or 'double'. 'single' means a single newline character, while 'double' means two consecutive newline characters. (default: double, works for most ebooks but will detect less paragraphs for some ebooks)",
+    )
+    parser.add_argument(
+        "--chapter_start",
+        default=1,
+        type=int,
+        help="Chapter start index (default: 1, starting from 1)",
+    )
+    parser.add_argument(
+        "--chapter_end",
+        default=-1,
+        type=int,
+        help="Chapter end index (default: -1, meaning to the last chapter)",
+    )
+    parser.add_argument(
+        "--output_text",
+        action="store_true",
+        help="Enable Output Text. This will export a plain text file for each chapter specified and write the files to the output folder specified.",
+    )
+    parser.add_argument(
+        "--remove_endnotes",
+        action="store_true",
+        help="This will remove endnote numbers from the end or middle of sentences. This is useful for academic books.",
+    )
+
+    parser.add_argument(
+        "--voice_name",
+        help="Various TTS providers has different voice names, look up for your provider settings.",
+    )
+    parser.add_argument(
```

Comment:
I agree with you, but here is the thing:

as I made Azure and OpenAI re-use some of the options - there is nearly nothing left for separate groups. 
For example for OpenAI - it would be only 1 param - `model_name` which would not match to Azure
Is it worth creating an entire group because of a single parameter? I decided not to. Though if there will be more and more TTS providers possibly we can group some of them in the future

So I'll skip this for now

Reply:
OK.